### PR TITLE
feat(elbv2): add metrics to INetworkTargetGroup and IApplicationTargetGroup

### PR DESCRIPTION
**Pull request by edisongustavo**
This PR follows the same conventions as #23853

By moving the metrics methods to the `INetworkTargetGroup` and `IApplicationTargetGroup` interfaces it allows to create these metrics also for Target Groups that are imported via the `fromTargetGroupAttributes()` method.

To create the metrics for Target Groups requires (1) the full name of the Target Group and (2) the full name of the Load Balancer.

For (1): it is readily available given that all imported Target Groups need to provide its ARN.
For (2), it is an optional value, so the `.metrics` parameter will throw an error if it was not provided.

To solve this problem I did:

- Introduce a new interface for each TG type: `INetworkTargetGroupMetrics`, `IApplicationTargetGroupMetrics`
- Create a concrete implementation for the new interfaces (1 for each): `NetworkTargetGroupMetrics` and `ApplicationTargetGroupMetrics`
- Make each concrete implementation of each Load Balancer to also provide a `metrics` field. The concrete implementations of the load balancers are:
`ImportedApplicationTargetGroup`, and `ApplicationLoadBalancer` (and the same for the NLB classes).

I chose to create a new interface because code can be reused across the 3 concrete implementations of each Load Balancer. I deprecated the `metricXXX()` methods of each load balancer because I think it is cleaner to access metrics through the new `metrics` attribute/interface.

There is a small **gotcha** here because the parameter of the `fromTargetGroupAttributes()` method that refers to the LB is: `loadBalancerArns`, which has its documentation as:

> A Token representing the list of ARNs for the load balancer routing to this target group

I'm not treating this parameter as a collection of ARNs, but as a single ARN. Also, I'm not treating it only as a token, but hardcoded ARNs can also be supplied, which "sort of" violates its interface. This attribute is weird though because Target Groups cannot have multiple Load Balancers as of today, although its documentation doesn't clearly express that is the case.

fix: #10850

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by edisongustavo** [comment]
This PR contains the commit from https://github.com/aws/aws-cdk/pull/23972, so that PR needs to be merged first.

**Comment by edisongustavo** [comment]
This PR contains the commit from https://github.com/aws/aws-cdk/pull/23972, so that PR needs to be merged first.

**Comment by edisongustavo** [comment]
This PR contains the commit from https://github.com/aws/aws-cdk/pull/23972, so that PR needs to be merged first.

File: packages/@aws-cdk/aws-elasticloadbalancingv2/test/alb/target-group.test.ts [hunk 1]
```diff
@@ -609,4 +609,39 @@ describe('tests', () => {
       },
     });
   });
+
+
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-elasticloadbalancingv2/test/alb/target-group.test.ts:613]:
```suggestion
```

**Review by comcalvi — CHANGES_REQUESTED** [review]
minor style comment, but other than that looks ready to merge!

**Review by comcalvi — APPROVED** [review]

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**